src/Microsoft.Azure.SignalR/ServiceRouteBuilder.cs

```diff
@@ -58,6 +59,18 @@ public ServiceRouteBuilder(RouteBuilder routes)
 
             var dispatcher = _serviceProvider.GetRequiredService<ServiceHubDispatcher<THub>>();
             dispatcher.Start(app);
+
+#if NETCOREAPP
+            var lifetime = _serviceProvider.GetService<IHostApplicationLifetime>();
+#elif NETSTANDARD
+            var lifetime = _serviceProvider.GetService<IApplicationLifetime>();
+#else
+            var lifetime = null;
+#endif
+            if (lifetime != null)
+            {
+                lifetime.ApplicationStopping.Register(() => dispatcher.ShutdownAsync(TimeSpan.FromSeconds(30)).Wait());

```

Comment:
>TimeSpan.FromSeconds(30) [](start = 85, length = 24)

timeout option is not respected?

Reply:
`docs` and `options` will be in the final PR

Reply:
then remove this line. Otherwise when your PR is in, the SDK contains this feature?